core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -340,6 +347,10 @@ public void onLeadershipChange(Set<Partition> partitionsBecomeLeader,
         Set<TopicIdPartition> followerPartitions = filterPartitions(partitionsBecomeFollower)
                 .map(p -> new TopicIdPartition(topicIds.get(p.topic()), p.topicPartition())).collect(Collectors.toSet());
 
+        if (this.rlmConfig.enableRemoteStorageSystem() && !isRemoteLogManagerConfigured()) {
+            throw new KafkaException("RemoteLogManager is not configured when remote storage system is enabled");
+        }
+
```

Comment:
nit: Should we move this validation on the top of `onLeadershipChange` method?

Reply:
Agree, we can throw before deriving other vars. Moved.